### PR TITLE
feat: settings page (resolves #128)

### LINKED ISSUE #128: Add privacy settings
**Is your feature request related to a problem? Please describe.**

Add privacy settings.

**Describe the solution you'd like**

Not applicable.

**Describe alternatives you've considered**

Not applicable.

**Additional context**

Not applicable.

### PR DESCRIPTION
**Pull request by greatislander**
* [x] I've read the [guidelines for Contributing to the Platform Co-op Toolkit](https://github.com/platform-coop-toolkit/.github/blob/master/CONTRIBUTING.md)
* [x] This isn't a duplicate of an existing pull request

## Description

This PR adds a settings page and gives users the ability to opt-in (via a cookie) to having their recently viewed resources tracked via Koko Analytics. If the `track_viewed_resources` cookie is set to `on`, the [`koko_analytics_load_tracking_script` filter](https://github.com/ibericode/koko-analytics/blob/6119cd5a789b3d6d4f24d2e6640d2724fe450b47/readme.txt#L108) is set to `true`; under all other circumstances it is set to false.

## Steps to test

1. Create a page called 'Settings' and set its page template to 'Settings'.
2. Visit this page, change the configuration to 'Allow', save.

**Expected behavior:** The `track_viewed_resources` cookie is set to true and subsequent visits to the settings page reflect this. When the `track_viewed_resources` cookie is set to true, subsequently browsed resources are added to the `_koko_analytics_pages_viewed` cookie.

## Additional information

Not applicable.

## Related issues

- Resolves #128.

### CODE REVIEW

File: resources/views/partials/content-page-settings.blade.php [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
-    'orderby' => 'menu_order',
-    'order' => 'ASC',
-    'post_parent' => $post->ID
-    ]) as $child)
-  <li class="link-list__item"><a href="{{ get_permalink($child->ID) }}">{{ $child->post_title }}</a></li>
-  @endforeach
-</ul>
+<form method="post">
+  <p><strong>{{ __('Allow the Resource Library to track what resources you’ve viewed.', 'coop-library') }}</strong></p>
+  <span class="disclosure-label" hidden>{{ __('How this changes what I see', 'coop-library') }}</span>
```
Comment (jhung) [inline resources/views/partials/content-page-settings.blade.php]: We may want to tweak the copy for this label. The word "this" may not be specific enough.

Perhaps "How resource tracking changes what I see"? @cherylhjli @dayotte  any thoughts?

File: resources/views/partials/content-page-settings.blade.php [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+  <div class="details">
+      <p>{{ __('If you allow the resource library to know this, you would be able to see resources that are recently viewed by you, related to your recently viewed, and contribute your information to create the most viewed resource section seen by the larger community.', 'coop-library') }}</p>
```
Comment (jhung) [inline resources/views/partials/content-page-settings.blade.php]: Suggestion to tweak to copy. I'm afraid it is getting longer, but it seems like it should be addressed whenever user tracking is involved:

"By choosing 'Allow' the Resource Library to track the resources you view, you will:
* see a list of your recently viewed resources on the home page,
* see related resources, and
* contribute to the most viewed articles seen on the home page.

The Resource Library will only use this information for these features, and does not collect any personally identifiable information.

By choosing 'Don't allow', you will be able to continue to use the Resource Library fully, with the exception of the features mentioned previously."

@cherylhjli @dayotte have any input?

File: app/setup.php [hunk 3]
```diff
@@ -96,6 +96,21 @@
 add_action('wp_ajax_update_favorites', 'App\\update_favorites');
 add_action('wp_ajax_nopriv_update_favorites', 'App\\update_favorites');
 
+/**
+ * Set language cookies when settings form is submitted.
+ */
+add_action('init', function () {
+    if (isset($_POST['track_viewed_resources'])) {
```
Comment (jhung) [inline app/setup.php]: When the form is submitted, a "Settings successfully saved" confirmation notification should appear.

File: resources/views/partials/content-page-settings.blade.php (hunk 2 above)
Comment (cherylhjli) [reply resources/views/partials/content-page-settings.blade.php]: this sounds good Jon!

File: app/setup.php (hunk 3 above)
Comment (cherylhjli) [reply app/setup.php]: sounds good too!

**Pull request merged**